shopinvader_delivery_carrier/tests/test_delivery_service.py

```diff
@@ -0,0 +1,228 @@
+        result = self.service.dispatch("search")
+        data = result.get("data", [])
+        self.assertFalse(data)
+        # Then create a picking related to the partner
+        picking = self._create_picking(partner=self.service.partner, sale=True)
+        self.assertEquals(picking.partner_id, self.service.partner)
+        result = self.service.dispatch("search")
+        data = result.get("data", [])
+        self._check_data_content(data, picking)
+        # Write some optional fields and re-check
+        self._fill_picking_optional_values(picking)
+        result = self.service.dispatch("search")
+        data = result.get("data", [])
+        self._check_data_content(data, picking)
+        return
+
+    def _fill_picking_optional_values(self, pickings):
+        """
+        Fill some optional values for given pickings
+        :param pickings: stock.picking
+        :return: bool
+        """
+        for picking in pickings:
+            picking.write(
+                {
+                    "carrier_id": self.carrier.id,
+                    "carrier_tracking_ref": str(uuid4()),
+                }
+            )
+        return True
+
+    def test_get_picking_logged_with_sale(self):
+        """
+        Test the get on a logged user.
+        In the first part, the user should have any picking.
+        But to the second, he should have one.
+        :return:
+        """
+        # Check first without picking related to the partner
+        result = self.service.dispatch("search")
+        data = result.get("data", [])
+        self.assertFalse(data)
+        # Then create a picking related to partner
+        picking = self._create_picking(partner=self.service.partner, sale=True)
+        self.assertEquals(picking.partner_id, self.service.partner)
+        result = self.service.dispatch("search")
+        data = result.get("data", [])
+        self._check_data_content(data, picking)
+        # Write some optional fields and re-check
+        self._fill_picking_optional_values(picking)
+        result = self.service.dispatch("search")
+        data = result.get("data", [])
+        self._check_data_content(data, picking)
+        return
+
+    def test_get_multi_picking(self):
+        """
+        Test the get on a logged user.
+        In the first part, the user should have any picking.
+        But to the second, he should have one.
+        :return:
+        """
+        picking1 = self._create_picking(
+            partner=self.service.partner, sale=True
+        )
+        picking2 = self._create_picking(
+            partner=self.service.partner, sale=True
+        )
+        picking3 = self._create_picking(
+            partner=self.service.partner, sale=True
+        )
+        picking4 = self._create_picking(
+            partner=self.service.partner, sale=True
+        )
+        pickings = picking1 | picking2 | picking3 | picking4
+        self.assertEquals(picking1.partner_id, self.service.partner)
```

Comment:
these assertions seem superfluous as you've just created the pickings w/ this partner